rust_src/src/process.rs

```diff
@@ -537,4 +538,23 @@ pub fn list_system_processes_rust() -> LispObject {
     unsafe { list_system_processes() }
 }
 
+/// Interrupt process PROCESS
+/// PROCESS may be a process, a buffer, or the name of a process or buffer.
+/// No arg or nil means current buffer's process.
+/// Second arg CURRENT-GROUP non-nil means send signal to
+/// the current process-group of the process's controlling terminal
+/// rather than to the process's own process group.
+/// If the process is a shell, this means interrupt current subjob
+/// rather than the shell.
+///
+/// If CURRENT-GROUP is `lambda', and if the shell owns the terminal,
+/// don't send the signal.
+///
+/// This function calls the functions of `interrupt-process-functions' in
+/// the order of the list, until one of them returns non-`nil'.
+#[lisp_fn(min = "0")]
+pub fn interrupt_process(process: LispObject, current_group: LispObject) -> LispObject {
+    run_hook_with_args_until_success(&mut [process, current_group])
```

Comment:
What happened to `Qinterrupt_process_functions`?

Reply:
You can port this from C to rust. Remove the `DEFSYM` in the C code and add `def_lisp_sym!()` in the Rust code. There are examples in eval.rs.